File: aws-net-workshop/aws-net-workshop/examples/_05_CreateObjectWithMetadata.cs
using System;
using Amazon.S3;
using Amazon.S3.Model;

namespace aws_net_workshop.examples
{
    class _05_CreateObjectWithMetadata
    {
        public static void Main(string[] args)
        {
            // create the AWS S3 Client
            AmazonS3Client s3 = AWSS3Factory.getS3Client();

            // retrieve the object key and object content from the user
            Console.Write("Enter the object key: ");
            string key = Console.ReadLine();
            Console.Write("Enter the object content: ");
            string content = Console.ReadLine();

            // retrieve the object metadata key and value from user
            Console.Write("Enter the metadata key: ");
            string metaKey = Console.ReadLine();
            Console.Write("Enter the metadata value: ");
            string metaValue = Console.ReadLine();

            // create object request with retrieved input
            PutObjectRequest request = new PutObjectRequest()
            {
                BucketName = AWSS3Factory.S3_BUCKET,
                ContentBody = content,
                Key = key
            };

            // add metadata to request
            request.Metadata.Add(metaKey, metaValue);

            // create the object with metadata in the demo bucket
            s3.PutObject(request);

            // print out object key/value and metadata key/value for validation
            Console.WriteLine(string.Format("Create object {0}/{1} with metadata {2}={3} and content: {4}", 
                AWSS3Factory.S3_BUCKET, key, metaKey, metaValue, content));
            Console.ReadLine();
        }
    }
}

File: aws-java-workshop/src/main/java/com/emc/ecs/s3/sample/_05_CreateObjectWithMetadata.java
/*
 * Copyright 2013-2018 EMC Corporation. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.txt
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
package com.emc.ecs.s3.sample;

import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.model.ObjectMetadata;
import com.amazonaws.util.StringInputStream;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class _05_CreateObjectWithMetadata {

    public static void main(String[] args) throws Exception {
        // create the AWS S3 Client
        AmazonS3 s3 = AWSS3Factory.getS3ClientWithV2Signatures();

        // retrieve the object key and value from user
        System.out.println( "Enter the object key:" );
        String key = new BufferedReader( new InputStreamReader( System.in ) ).readLine();
        System.out.println( "Enter the object content:" );
        String content = new BufferedReader( new InputStreamReader( System.in ) ).readLine();

        //retrieve the object metadata key and value from user
        System.out.println( "Enter the metadata key:" );
        String metaKey = new BufferedReader( new InputStreamReader( System.in ) ).readLine();
        System.out.println( "Enter the metadata content:" );
        String metaValue = new BufferedReader( new InputStreamReader( System.in ) ).readLine();

        // create the metadata
        ObjectMetadata metadata = new ObjectMetadata();
        metadata.addUserMetadata(metaKey, metaValue);

        // Set the content type for streaming back to a browser.
        metadata.setContentType("text/plain");

        // create the object with the metadata in the demo bucket
        s3.putObject(AWSS3Factory.S3_BUCKET, key, new StringInputStream(content), metadata);

        // print out object key/value and metadata key/value for validation
        System.out.println( String.format("created object [%s/%s] with metadata [%s=%s] and content: [%s]",
                AWSS3Factory.S3_BUCKET, key, metaKey, metaValue, content));
    }
}
